sdk/core/core-http/src/url.ts

```diff
@@ -200,7 +200,7 @@ export class URLBuilder {
    * path or query), those parts will be added to this URL as well.
    */
   public setPort(port: number | string | undefined): void {
-    if (port == undefined || port === "") {
+    if (port === undefined || port === "") {
```

Comment:
Similar to the feedback in https://github.com/Azure/azure-sdk-for-js/pull/8687/files#r422756074, either dont change `== undefined`  to `=== undefined` and have the library owners follow up or replace `x == undefined` with `x === undefined || x=== null` . 

Same for other places where the same change is being made

Reply:
Done!